pkg/trace/agent/agent.go

```diff
@@ -105,15 +122,15 @@ func NewAgent(ctx context.Context, conf *config.AgentConfig, telemetryCollector
 	dynConf := sampler.NewDynamicConfig()
```

Comment:
We can probably do this in a later PR,  but we should consider passing in the concentrator, writer, etc. to make it easier for OTel to build an agent with the components they want.

We may be able to refactor the component code as well to use an actual agent instead of the agent wrapper they use there.